diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/BracesInConditionalsAndLoopsRule.kt

```diff
@@ -0,0 +1,157 @@
+
+class BracesInConditionalsAndLoopsRule : Rule("braces-rule") {
+    companion object {
+        private const val indentStep = 4
+    }
+
+    private lateinit var configRules: List<RulesConfig>
+    private lateinit var emitWarn: ((offset: Int, errorMessage: String, canBeAutoCorrected: Boolean) -> Unit)
+    private var isFixMode: Boolean = false
+
+    override fun visit(node: ASTNode,
+                       autoCorrect: Boolean,
+                       params: KtLint.Params,
+                       emit: (offset: Int, errorMessage: String, canBeAutoCorrected: Boolean) -> Unit) {
+        configRules = params.getDiktatConfigRules()
+        emitWarn = emit
+        isFixMode = autoCorrect
+
+        when (node.elementType) {
+            IF -> checkIfNode(node)
+            WHEN -> checkWhenBranches(node)
+            FOR, WHILE, DO_WHILE -> checkLoop(node)
+        }
+    }
+
+    /**
+     * Check braces in if-else statements. Check for both IF and ELSE needs to be done in one method to discover single-line if-else statements correctly.
+     * There is KtIfExpression class which can be used to access `then` and `else` body.
+     */
+    @Suppress("ForbiddenComment")
+    private fun checkIfNode(node: ASTNode) {
+        val ifPsi = node.psi as KtIfExpression
+        val thenNode = ifPsi.then?.node
+        val hasElseBranch = ifPsi.elseKeyword != null
+        val elseNode = ifPsi.`else`?.node
+        val indent = node.prevSibling { it.elementType == WHITE_SPACE }?.text?.lines()?.last()?.count { it == ' ' } ?: 0
+
+        if (isSingleLineIfElse(node, elseNode)) return
+
+        if (thenNode?.elementType != BLOCK) {
+            NO_BRACES_IN_CONDITIONALS_AND_LOOPS.warnAndFix(configRules, emitWarn, isFixMode, "IF",
+                    (thenNode?.prevSibling { it.elementType == IF_KEYWORD } ?: node).startOffset) {
+                if (thenNode != null) {
+                    ifPsi.then!!.replaceWithBlock(indent)
+                    if (elseNode != null) {
+                        node.replaceChild(node.findChildByType(ELSE_KEYWORD)!!.treePrev, PsiWhiteSpaceImpl(" "))
+                    }
+                } else {
+                    val nodeAfterCondition = node.findChildByType(CONDITION)!!.nextSibling { it.elementType == ElementType.RPAR }!!.treeNext
+                    node.insertEmptyBlockBetweenChildren(nodeAfterCondition, nodeAfterCondition, indent)
+                }
+            }
+        }
+
+        if (hasElseBranch && elseNode?.elementType != IF && elseNode?.elementType != BLOCK) {
+            NO_BRACES_IN_CONDITIONALS_AND_LOOPS.warnAndFix(configRules, emitWarn, isFixMode, "ELSE",
+                    (elseNode?.treeParent?.prevSibling { it.elementType == ELSE_KEYWORD } ?: node).startOffset) {
+                if (elseNode != null) {
+                    ifPsi.`else`!!.replaceWithBlock(indent)
+                } else {
+                    // fixme: in which case else can have empty body?
```

Comment:
simply someone added empty branch? 
for `when` we even have such rule in code style (for when we should always have else branch)

Reply:
Yes, but I couldn't come up with example which can be compiled with empty `else` branch for `if` statement. If you do something like 

```
if (condition)
    foo()
else
```

compiler will complain that expression is expected.